tests/by-util/test_du.rs

```diff
@@ -580,97 +580,62 @@ fn test_du_invalid_threshold() {
 
 #[test]
 fn test_du_apparent_size() {
-    let ts = TestScenario::new(util_name!());
-    let result = ts.ucmd().arg("--apparent-size").succeeds();
+    let (at, mut ucmd) = at_and_ucmd!();
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    at.mkdir_all("a/b");
+    at.touch("a/b/file1");
+    at.touch("a/b/file2");
+
+    at.write("a/b/file1", "foo");
+    at.write("a/b/file2", "foobar");
+
+    let result = ucmd.args(&["--apparent-size", "--all", "a"]).succeeds();
+
+    #[cfg(not(target_os = "windows"))]
     {
-        let result_reference = unwrap_or_return!(expected_result(&ts, &["--apparent-size"]));
-        assert_eq!(result.stdout_str(), result_reference.stdout_str());
+        result.stdout_contains("1\ta/b/file1");
+        result.stdout_contains("1\ta/b/file2");
+        result.stdout_contains("1\ta/b");
+        result.stdout_contains("1\ta");
```

Comment:
I think the patterns should end with `\n` as I guess you want to verify whether there are lines containing "1\ta/b" and "1\ta".